pkg/controller/persistentvolume/persistentvolume_provisioner_controller.go

```diff
@@ -0,0 +1,314 @@
+	provisioner.lock.Lock()
+	defer provisioner.lock.Unlock()
+	volume := newObj.(*api.PersistentVolume)
+	if claimKey, exists := volume.Annotations[createdForKey]; exists {
+		provisioner.provisionedClaims[claimKey] = true
+	}
+}
+
+func (provisioner *PersistentVolumeProvisioner) deleteVolume(obj interface{}) {
+	provisioner.lock.Lock()
+	defer provisioner.lock.Unlock()
+	volume := obj.(*api.PersistentVolume)
+	if claimKey, exists := volume.Annotations[createdForKey]; exists {
+		delete(provisioner.provisionedClaims, claimKey)
+	}
+}
+
+func (provisioner *PersistentVolumeProvisioner) addClaim(obj interface{}) {
+	provisioner.lock.Lock()
+	defer provisioner.lock.Unlock()
+	claim := obj.(*api.PersistentVolumeClaim)
+	if _, exists := claim.Annotations[provisionedKey]; exists {
+		provisioner.provisionedClaims[ClaimToProvisionableKey(claim)] = true
+	}
+	err := provisionClaim(claim, provisioner.creater, provisioner.client, provisioner.provisionedClaims)
+	if err != nil {
+		glog.Errorf("PersistentVolumeProvisioner could not handle claim %s: %+v", claim.Name, err)
+	}
+}
+
+func (provisioner *PersistentVolumeProvisioner) updateClaim(oldObj, newObj interface{}) {
+	provisioner.lock.Lock()
+	defer provisioner.lock.Unlock()
+	claim := newObj.(*api.PersistentVolumeClaim)
+	if _, exists := claim.Annotations[provisionedKey]; exists {
+		provisioner.provisionedClaims[ClaimToProvisionableKey(claim)] = true
+	}
+	err := provisionClaim(claim, provisioner.creater, provisioner.client, provisioner.provisionedClaims)
+	if err != nil {
+		glog.Errorf("PersistentVolumeProvisioner could not handle claim %s: %+v", claim.Name, err)
+	}
+}
+
+func provisionClaim(pvc *api.PersistentVolumeClaim, plugin volume.CreatableVolumePlugin, client provisionerClient, provisionedClaims map[string]bool) error {
+	// TODO: revisit this when field selectors can look for specific annotations
+	if _, exists := pvc.Annotations[provisionableKey]; !exists {
+		glog.V(5).Infof("PersistentVolumeClaim[%s/%s] missing provisionable annotation from binder.  No volume will be created.", pvc.Namespace, pvc.Name)
+		return nil
+	}
+
+	if _, exists := pvc.Annotations[provisionedKey]; exists {
+		glog.V(5).Infof("PersistentVolumeClaim[%s/%s] has already been provisioned.  No volume will be creatd.", pvc.Namespace, pvc.Name)
+		return nil
+	}
+
+	if _, exists := provisionedClaims[ClaimToProvisionableKey(pvc)]; exists {
+		glog.V(5).Infof("PersistentVolumeClaim[%s/%s] has already been provisioned.  No volume will be creatd.", pvc.Namespace, pvc.Name)
+		return nil
+	}
+
+	qos, exists := pvc.Annotations[qosProvisioningKey]
+	if !exists {
+		return fmt.Errorf("PersistentVolumeClaim[%s/%s] missing quality of service annotation", pvc.Namespace, pvc.Name)
+	}
+	glog.V(5).Infof("Provisioning PersistentVolumeClaim[%s/%s]\n", pvc.Namespace, pvc.Name)
+
+	volumeOptions := volume.VolumeOptions{
+		CapacityMB:                    100,
```

Comment:
The volume size must come from the claim, otherwise it creates only 100MB volumes.